### DISCUSSION TITLE
FDQuery LoadFromJson

### DISCUSSION THREAD

**Discussion by pabloanizio** [viniciussanchez/dataset-serialize#235] [General]
Olá.

Quando eu carrego um json com os dados de um registro pelo LoadFromJson, ele está desconsiderando os dados dos campos autoinc (id). Por exemplo: no json eu tenho {"id":1,"nome":"Teste"}, no FDQuery, após abrir, ele fica assim: id: -1, nome: Teste.

Tem alguma forma de eu contornar esse comportamento?

**Comment by viniciussanchez** [comment]
Bom dia... a princípio, se o campo é autoinc, deixe ele como ReadOnly e o mesmo será ignorado na importação. Outra possibilidade também, é deixar o Field como Visible = False.

**Comment by pabloanizio** [comment]
Bom dia Vinícius. Obrigado pela resposta. Eu fui reler o readme e vi no que eu estava errando. Eu estava tentando trabalhar da mesma forma do dataset.loadfromfile, onde é enviado bem mais informações  no arquivo e por isso o dataset é montado corretamente. Na forma de trabalhar como o dataset.loadfromjson, o dataset já deve está com os dados do registro carregados, para só assim o método aplicar as diferenças.
